### ISSUE TITLE
Extension causes high cpu load

### ISSUE THREAD

**Issue by spencer-shadley** [prettier/prettier-vscode#1201]
- Issue Type: `Performance`
- Extension Name: `prettier-vscode`
- Extension Version: `3.18.0`
- OS Version: `Darwin x64 18.7.0`
- VSCode version: `1.41.1`

Find more details here: https://github.com/microsoft/vscode/wiki/Explain-extension-causes-high-cpu-load

[esbenp.prettier-vscode-unresponsive.cpuprofile.txt](https://github.com/prettier/prettier-vscode/files/4069516/esbenp.prettier-vscode-unresponsive.cpuprofile.txt)

**Comment by MrEfrem** [comment]
For me it's big memory usage yet.

**Comment by MrEfrem** [comment]
For me reason was that my workspace contains many projects with own prettier instances in `node_modules` therefore when I edited files in all them this extension consumed almost all resources my PC (about 7Gb memory and almost always 100% CPU). But then I installed `prettier` globally, I've set `Prettier Path` to it and then all became work very well. Now VSCode takes about 1Gb memory and there isn't more the high cpu load.

**Comment by ntotten** [comment]
See #1135. I think this proposal would fix this problem for large repos.

**Comment by daveblee** [comment]
I'm still seeing this issue.
[esbenp.prettier-vscode-unresponsive.cpuprofile.txt](https://github.com/prettier/prettier-vscode/files/4612423/esbenp.prettier-vscode-unresponsive.cpuprofile.txt)

**Comment by gianpaj** [comment]
```
Version: 1.45.1
Commit: 5763d909d5f12fe19f215cbfdd29a91c0fa9208a
Date: 2020-05-14T08:33:47.663Z (2 wks ago)
Electron: 7.2.4
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Darwin x64 19.5.0
```
[esbenp.prettier-vscode-unresponsive.cpuprofile.txt](https://github.com/prettier/prettier-vscode/files/4716469/esbenp.prettier-vscode-unresponsive.cpuprofile.txt)

**Comment by ntotten** [comment]
As an FYI, the most common reason for these perf issues is that you open a project with lots of subprojects that each have their own version of prettier. What you can do is set the [`prettierPath`](https://github.com/prettier/prettier-vscode#prettierprettierpath) to force the extension to use a single instance of prettier for your entire project. 

Something like `./node_modules/prettier` will set it to always use the module at the root of your workspace.

**Comment by ntotten** [comment]
Please try out the preview release of the Prettier extension. There have been a number of performance changes that I believe should help with this. https://github.com/prettier/prettier-vscode/wiki/Testing-Preview-Releases

I don't want to release this to the main channel until we get a decent amount of testers to try it out because the changes are significant. Please report any bugs and let me know if you see performance improvements.

**Comment by lukecotter** [comment]
@ntotten I just tried the beta the startup activation time was 47ms instead of 545ms with the release version.
I also had `"prettier.prettierPath": "./node_modules/prettier",` set.

**Comment by ntotten** [comment]
Should be resolved in v6.0
